packages/opentelemetry-plugin-grpc/src/grpc.ts

```diff
@@ -329,17 +329,19 @@ export class GrpcPlugin extends BasePlugin<grpc> {
     client: typeof grpcTypes.Client,
     methods: { [key: string]: { originalName?: string } }
   ): string[] {
-    const methodsToWrap = [
-      ...Object.keys(methods),
-      ...(Object.keys(methods)
-        .map(methodName => methods[methodName].originalName)
-        .filter(
-          originalName =>
-            // eslint-disable-next-line no-prototype-builtins
-            !!originalName && client.prototype.hasOwnProperty(originalName)
-        ) as string[]),
-    ];
-    return methodsToWrap;
+    const methodSet = new Set<string>(); // use set to remove duplicates caused by "camelCase"
```

Comment:
Just trying to understand it. Can you please explain shortly what has been returned previously and how it looks now ?

Reply:
This returns a list of strings which are methods to patch. Each method has a `name` and a `originalName`. There were some cases where this caused certain method names to be duplicated in the returned array. Using a Set, we remove all duplicates.

Reply:
Sure. If someone's `.proto` has a method starting with an uppercase character,
```proto
rpc UpperCase ( foo ) returns ( foo );
```
`_getMethodsToWrap` is called with `methods = { upperCase: { originalName: "UpperCase", ... }}`.

GRPC will ultimately create 2 client methods for the user called `client.UpperCase(...)` and `client.upperCase(...)`. They are separate functions, and so they both need to be patched separately. We grab the alternative function name from `originalName`. The expected return here is `["upperCase", "UpperCase"]`. This worked correctly for this `.proto` use case.

If someone's `.proto` has a method like this
```proto
rpc lowerCase ( foo ) returns ( foo );
```

GRPC will create only 1 client method -- `client.lowerCase(...)`, but `originalName` will still exist, `methods = { lowerCase: { originalName: "lowerCase", ... }}`. Prior to this PR, this function would return `["lowerCase", "lowerCase"]` and double patch the functions.

Looking at this again, I think I can forgo the `Set` and just add an extra equality check to `methodList.push(originalName)` and get similar / better performance compared to using a Set.

Reply:
Ah yea, I agree a set is not the best way to handle that. An equality check is likely a better option.